### PR TITLE
Fix contracts indexer

### LINKED ISSUE #97: Fix contracts indexer
Issues:

- enable pre byzantium contracts by removing redundant receipt status check (the archive only stores successfully created contracts as it uses the contract address property on the receipt to detect the transactions)
- fix non-detection of ERC1155 contracts
- fix non-detection of Uniswap v3 contracts

### PR DESCRIPTION
**Pull request by bdmason**
Closes #97 

- enable pre byzantium contracts by removing redundant receipt status check (the archive only stores successfully created contracts as it uses the contract address property on the receipt to detect the transactions)
- fix non-detection of ERC1155 contracts
- fix non-detection of Uniswap v3 contracts

Also:
- reworks the indexer to a more standard ETL code split
- extraction now batches requests to stop the max response size errors
- load now uses upsert, and writes batches rather than individual recods

### CODE REVIEW

File: indexers/contracts/src/indexer/extract.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+          'blockTimestamp',
+          'input',
+        ],
+      });
+    }),
+  );
+
+  const txs = results.flatMap((txArr) => txArr);
+
+  return txs;
```
Comment (jdomingos) [inline indexers/contracts/src/indexer/extract.ts:27]: For simplicity we could do:

```suggestion
  return results.flatMap((txArr) => txArr);
```

File: indexers/contracts/src/indexer/transform.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
+import { utils, Types } from 'indexer-utils';
+import fetchTokenData from './fetch-token-data';
+import { TransformResult, Model } from './types';
+
+export default async function transform(
```
Comment (jdomingos) [inline indexers/contracts/src/indexer/transform.ts:5]: Minor: I've noticed we don't have metrics here, so we don't see anything on the Contracts Dashboard

**Review by jdomingos — APPROVED** [review]
Looks good, nicely done! 

Just a couple of comments - one suggestion and one thing that I noticed about metrics

File: indexers/contracts/src/indexer/transform.ts (hunk 2 above)
Comment (bdmason) [reply indexers/contracts/src/indexer/transform.ts:5]: the metrics should be coming from the parent file `indexer/index.ts`

```
  console.time('transform');
  monitoring.markStart(metrics.transformBlock);
  const transformed = await transform(txs);
  monitoring.markEnd(metrics.transformBlock);
  console.timeEnd('transform');
```

**Pull request merged**